internal/pipe/docker/docker.go

```diff
@@ -61,8 +61,10 @@ func (Pipe) Default(ctx *context.Context) error {
 		if err := validateImager(docker.Use); err != nil {
 			return err
 		}
+		distPath := filepath.Clean(ctx.Config.Dist) + "/"
```

Comment:
I think we should probably use os.Separator instead of `/` 🤔

Reply:
another idea is to try to use `filepath.Rel` to check if one path is within dist or not 🤔

Reply:
I started to use the filepath.Rel() function. I couldn't figure out how to write the if statement if using it. 

```
filepath.Rel(ctx.Config.Dist, ".") == ../../.
filepath.Rel(ctx.Config.Dist, "./tmp/dist") == "."
filepath.Rel(ctx.Config.Dist, "./tmp/dist/something") == "something"
filepath.Rel(ctx.Config.Dist, "./tmp/dist_files/something") == ../dist_files/something
```

Reply:
hmmm true, I guess your current solution is better...

do you think we should use os.separator instead of forward slash?

Reply:
Yes, I think so. 

Should it be possible to use like ".." as extras_file path?

Reply:
tbh I don't quite remember why we have this check... it might be some old thing, as once upon a time we symlinked the files, and I think this was to avoid circular linking... maaaaybe this check can be removed... 

do you want to test it out?

like, building a sample project adding files that would be refused in the current impl, and seeing what happens... I'm 95% sure it will work now...

Reply:
I just tested it. I removed the check and added this in my .goreleaser.yaml

```
#.goreleaser.yaml
dockers:
  - image_templates:
    - fgummesson/myrepo/extra-files-test:{{.Version}}
    extra_files:
      - "."
`
``


then ran 
```
 $> goreleaser release --snapshot --clean
  • starting release...
  • loading config file                              file=.goreleaser.yaml
.....
  • docker images
  ⨯ release failed after 0s                  error=docker build failed: failed to copy extra file '.': failed to open 'dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/Dockerfile': open dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/dist/goreleaserdocker2114537310/Dockerfile: file name too long
```

Reply:
Ahhh, yeah, it stills create the temp inside dist, maybe we can simply create it elsewhere... I'll take a look later 🙏

Reply:
indeed, this is not needed anymore https://github.com/goreleaser/goreleaser/pull/3823